src/main.rs

```diff
@@ -309,16 +315,28 @@ fn process_file(
     let mut include = vec![true];
     let mut first_line = true;
 
+    // Create a new Rhai engine and scope
+    let mut engine = rhai::Engine::new();
+    let mut scope = rhai::Scope::new();
+    scope.push("options", options.to_vec());
+
+    // Define a custom function to check if conditions of the options.
+    engine.register_fn(
+        "contains_option",
+        |cond: &str, options: Vec<String>| -> bool {
+            options.iter().any(|option| cond.contains(option))
+        },
+    );
+
```

Comment:
I think we can make it easier by registering this function:
```rust
    let options_clone: Vec<String> = options.iter().map(|v| v.to_owned()).collect();
    engine.register_fn(
        "contains_option",
        move |cond: &str| -> bool {
            let cond = cond.to_string();
            options_clone.contains(&cond)
        },
    );
```

and not doing any processing of the condition

Reply:
But then the template would need to have conditions like this `//IF contains_option("alloc")` - probably naming the function just `option` would be more convenient